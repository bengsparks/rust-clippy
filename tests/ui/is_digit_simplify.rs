#![warn(clippy::is_digit_simplify)]
fn main() {
    let c = '0';
    
    let _good = [
        c.is_digit(0),
        c.is_digit(1),
        c.is_digit(9),
        c.is_digit(11)
    ];

    let _bad = [
        c.is_digit(10),
        c.is_digit(16)
    ];
}
