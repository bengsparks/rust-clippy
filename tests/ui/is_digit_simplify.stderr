error: is_digit called with decimal radix
  --> $DIR/is_digit_simplify.rs:13:11
   |
LL |         c.is_digit(10),
   |           ^^^^^^^^^^^^
   |
   = note: `-D clippy::is-digit-simplify` implied by `-D warnings`
   = help: consider using is_ascii_digit

error: is_digit called with hexadecimal radix
  --> $DIR/is_digit_simplify.rs:14:11
   |
LL |         c.is_digit(16)
   |           ^^^^^^^^^^^^
   |
   = help: consider using is_ascii_hexdigit

error: aborting due to 2 previous errors

